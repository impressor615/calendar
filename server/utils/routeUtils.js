const errors = require('../errors');

module.exports = {
  sendError: ({
    res,
    language,
    error = errors.route_invalid_data,
    status = 400,
  }) => {
    const message = error[language] || error.en;
    return res.status(status).json({
      type: 'error',
      message,
    });
  },
};
