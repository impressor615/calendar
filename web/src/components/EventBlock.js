import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Button from 'components/Button';


const EventBlock = ({
  children,
  onToggle,
  onDragStart,
  fluid,
  ...rest
}) => (
  <Button
    {...rest}
    draggable
    onClick={onToggle}
    onDragStart={onDragStart}
    className={classnames(
      'event-block',
      rest.className,
      { fluid },
    )}
  >
    { children }
  </Button>
);

EventBlock.defaultProps = {
  fluid: false,
};

EventBlock.propTypes = {
  children: PropTypes.node.isRequired,
  onToggle: PropTypes.func.isRequired,
  onDragStart: PropTypes.func.isRequired,
  fluid: PropTypes.bool,
}

export default EventBlock;