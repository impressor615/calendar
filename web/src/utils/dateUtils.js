import moment from 'moment';

export const monthRange = (date) => {
  const startDate = moment(date).startOf('month').startOf('week');
  const endDate = moment(date).endOf('month').endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return {
    start_date: startDate,
    end_date: endDate,
    items: Array.from({ length: diff }).map((_, index) => {
      const targetDate = moment(startDate).add(index, 'days');
      return {
        year: targetDate.year(),
        month: (targetDate.month() + 1),
        date: targetDate.date(),
        moment: targetDate,
      };
    }),
  }
};

export const weekRange = (date) => {
  const startDate = moment(date).startOf('week');
  const endDate = moment(date).endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return {
    start_date: startDate,
    end_date: endDate,
    items: Array.from({ length: diff }).map((_, index) => {
      const targetDate = moment(startDate).add(index, 'days');
      return {
        year: targetDate.year(),
        month: (targetDate.month() + 1),
        date: targetDate.date(),
        moment: targetDate,
      };
    }),
  };
};

export const getHourKey = (hour) => {
  let hourKey = hour.toString();
  if (hour <= 9) {
    hourKey = `0${hourKey}`;
  }
  return hourKey;
};
