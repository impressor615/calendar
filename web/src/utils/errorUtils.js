import CONSTANTS from '../constants';

const { ERRORS } = CONSTANTS;
export const getErrorMsg = (error) => {
  const { response } = error.request;
  const errorObj = JSON.parse(response);
  const msg = ERRORS[errorObj.code].ko || ERRORS['401'].ko;
  return msg;
}