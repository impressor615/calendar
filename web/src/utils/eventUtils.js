export const eventBind = (onEvent, props) => event => onEvent(event, props);