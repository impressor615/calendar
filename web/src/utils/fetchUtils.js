import axios from 'axios';


export const authAxios = axios.create({ headers: { 'x-access-token': process.env.REACT_APP_SECRET } });

export const fetchEvents = (startDate, endDate) => {
  const start_date = startDate.toISOString();
  const end_date = endDate.toISOString();
  return authAxios.get('/api/calendar', { params: { start_date, end_date }})
    .then(res => res.data)
};

export const updateServerEvent = (_id, putData, events) => (
  authAxios.put(`/api/calendar/${_id}`, putData)
    .then(() => {
      const newEvents = [...events].filter(item => item._id !== _id);
      putData._id = _id
      newEvents.push(putData);
      return newEvents;
    })
);

