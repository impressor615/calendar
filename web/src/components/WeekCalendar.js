import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import EventBlock from 'components/EventBlock';
import { eventBind } from 'utils/eventUtils';
import CONSTANTS from '../constants';


const { DAY, HOURS } = CONSTANTS;
export const CalendarTime = ({
  children,
  onToggle,
  onDrop,
}) => (
  <div
    className="cal-block sm"
    onClick={onToggle}
    onDragOver={eventBind((e) => e.preventDefault())}
    onDrop={onDrop}
  >
    {children}
  </div>
);

CalendarTime.defaultProps = {
  children: '',
};

CalendarTime.propTypes = {
  children: PropTypes.node,
  onToggle: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired,
};

const WeekCalendar = ({
  range,
  onToggle,
  onDragStart,
  onDrop,
  currentDate,
  data,
}) => (
  <section className="week-calendar">
    <div className="cal-block sm" />
    {
      range.map((dateObj, index) => {
        const isYearShowing = currentDate.year() !== dateObj.year;
        const isMonthShowing = (currentDate.month() + 1) !== dateObj.month;
        return (
          <div className="cal-block sm" key={index.toString()}>
            <div className="day">{DAY[index]}</div>
            <div>
              { isYearShowing ? `${dateObj.year}년 ${dateObj.month}월 ${dateObj.date}일` : null }
              { !isYearShowing && isMonthShowing ? `${dateObj.month}월 ${dateObj.date}일` : null }
              { !isYearShowing && !isMonthShowing ? `${dateObj.date}` : null }
            </div>
          </div>
        );
      })
    }
    {
      HOURS.map((hourObj, index) => (
        <Fragment key={index.toString()}>
          <div className="cal-block sm">
            <span className="hour">{hourObj.text}</span>
          </div>
          {
            range.map((dateObj, index) => {
              const { text , code } = hourObj;
              const intCode = parseInt(code, 10);
              const blockDate = moment(dateObj.moment).hours(intCode).format('YYYY-MM-DD hh:mm A');
              const event = data.find((item) => {
                const eventDate = moment(item.start_date).format('YYYY-MM-DD hh:mm A');
                return eventDate === blockDate;
              });
              return (
                <CalendarTime
                  key={index.toString()}
                  onToggle={eventBind(onToggle, { type: 'week', hour: code, dateObj })}
                  onDrop={eventBind(onDrop, { type: 'week', hour: code, dateObj })}
                >
                  {
                    event
                      ? (
                        <EventBlock
                          fluid
                          className="secondary"
                          key={index.toString()}
                          onToggle={eventBind(onToggle, { type: 'event', dateObj: event })}
                          onDragStart={eventBind(onDragStart, { event })}
                        >
                          <div>{event.title}</div>
                          <div>{ text || '오전 00시' }</div>
                        </EventBlock>
                      ) : null
                  }
                </CalendarTime>
              );
            })
          }
        </Fragment>
      ))
    }
  </section>
);

WeekCalendar.propTypes = {
  range: PropTypes.arrayOf(PropTypes.object).isRequired,
  onToggle: PropTypes.func.isRequired,
  onDragStart: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired,
  currentDate: PropTypes.object.isRequired,
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default WeekCalendar;