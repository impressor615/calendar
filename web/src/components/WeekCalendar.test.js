import React from 'react';
import moment from 'moment';

import WeekCalendar, { CalendarTime } from 'components/WeekCalendar';
import EventBlock from 'components/EventBlock';
import { weekRange } from 'utils/dateUtils';


describe("<WeekCalendar />", () => {
  const onToggle = jest.fn();
  const onDrop = jest.fn();
  const onDragStart = jest.fn();
  const currentDate = moment('2018-10-14');
  const { range } = weekRange(currentDate);
  const props = {
    range,
    currentDate,
    onToggle,
    onDrop,
    onDragStart,
    data: [
      {
        title: 'title',
        start_date: moment(currentDate).toISOString(),
        end_date: moment(currentDate).add(1, 'hours').toISOString(),
      },
    ]
  };

  it('should be rendered properly', () => {
    const wrapper = shallow(<WeekCalendar {...props} />);
    expect(wrapper.find('.week-calendar')).toHaveLength(1);
    expect(wrapper.find(CalendarTime)).toHaveLength(24 * 7);
    expect(wrapper.find(EventBlock)).toHaveLength(1);
  });

  it('methods should be called properly', () => {
    const wrapper = shallow(<WeekCalendar {...props} />);
    wrapper.find(CalendarTime).first().simulate('drop');
    wrapper.find(EventBlock).simulate('dragstart');
    expect(onDrop).toHaveBeenCalledTimes(1);
    expect(onDragStart).toHaveBeenCalledTimes(1);
  });

  it('should be matched with the snapshot', () => {
    const wrapper = shallow(<WeekCalendar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
