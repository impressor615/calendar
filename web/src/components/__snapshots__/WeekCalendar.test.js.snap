// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeekCalendar /> should be matched with the snapshot 1`] = `
<section
  className="week-calendar"
>
  <div
    className="cal-block sm"
  />
  <div
    className="cal-block sm"
    key="0"
  >
    <div
      className="day"
    >
      일
    </div>
    <div>
      14
    </div>
  </div>
  <div
    className="cal-block sm"
    key="1"
  >
    <div
      className="day"
    >
      월
    </div>
    <div>
      15
    </div>
  </div>
  <div
    className="cal-block sm"
    key="2"
  >
    <div
      className="day"
    >
      화
    </div>
    <div>
      16
    </div>
  </div>
  <div
    className="cal-block sm"
    key="3"
  >
    <div
      className="day"
    >
      수
    </div>
    <div>
      17
    </div>
  </div>
  <div
    className="cal-block sm"
    key="4"
  >
    <div
      className="day"
    >
      목
    </div>
    <div>
      18
    </div>
  </div>
  <div
    className="cal-block sm"
    key="5"
  >
    <div
      className="day"
    >
      금
    </div>
    <div>
      19
    </div>
  </div>
  <div
    className="cal-block sm"
    key="6"
  >
    <div
      className="day"
    >
      토
    </div>
    <div>
      20
    </div>
  </div>
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    />
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  >
    <EventBlock
      className="secondary"
      fluid={true}
      key="0"
      onDragStart={[Function]}
      onToggle={[Function]}
    >
      <div>
        title
      </div>
      <div>
        오전 00시
      </div>
    </EventBlock>
  </CalendarTime>
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 1시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 2시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 3시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 4시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 5시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 6시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 7시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 8시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 9시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 10시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오전 11시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 12시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 1시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 2시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 3시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 4시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 5시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 6시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 7시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 8시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 9시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 10시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <div
    className="cal-block sm"
  >
    <span
      className="hour"
    >
      오후 11시
    </span>
  </div>
  <CalendarTime
    key="0"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="1"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="2"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="3"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="4"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="5"
    onDrop={[Function]}
    onToggle={[Function]}
  />
  <CalendarTime
    key="6"
    onDrop={[Function]}
    onToggle={[Function]}
  />
</section>
`;
